### PR TITLE
Event Feed Adding Data Lifecycle

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description
Adding the data lifecycle service to the event feed service. The data lifecycle will default in deleting events that are older than 7 days. Add `eventfeed-*` Elasticsearch indexes will be backed up. 

### :athletic_shoe: Demo Script / Repro Steps
1. Add an event older than 7 days; edit the `Published` date to be older than 7 days. 
1. Re-source the change with `source .studio/event-feed-service`
1. Add the event `event_feed_add_event`
1. List all the events to see the new event added with `event_feed_list`
1. Trigger the data lifecycle purge with `chef-automate dev grpcurl data-lifecycle-service -- chef.automate.domain.data_lifecycle.DataLifecycle.TriggerPurge`
1. List all the events again to see that the event was removed. 

### :chains: Related Resources

https://github.com/chef/automate/issues/444

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [x] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: .studio/golang [hunk 1]
```diff
@@ -184,6 +184,11 @@ function go_update_component() {
 # Adding auto tab complete
 complete -F _component_auto_complete go_update_component
 
+document "kill_running_service" <<DOC
+  kill a running component. This can be used to restart a component
+
+  example kill_running_service ingest-service
+DOC
```
Comment (lancewf) [inline .studio/golang]: Adding the `document` element to add this function to the hab studio describe function.

File: api/config/event_feed/config_request.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ func DefaultConfigRequest() *ConfigRequest {
 	c.V1.Sys.Service.Host = w.String("0.0.0.0")
 	c.V1.Sys.Service.Port = w.Int32(10134)
 	c.V1.Sys.Log.Level = w.String("info")
+	c.V1.Sys.Service.PurgeEventFeedAfterDays = w.Int32(7)
```
Comment (lancewf) [inline api/config/event_feed/config_request.go]: Adding the number of days to purge the event feed data to the Automate config.

File: api/config/event_feed/config_request.proto [hunk 3]
```diff
@@ -28,6 +28,7 @@ message ConfigRequest {
 			message Service {
 				google.protobuf.StringValue host = 1;
 				google.protobuf.Int32Value port = 2 [ (chef.automate.api.port) = { name: "event-feed-service", default: 10134, protocol: "grpc" } ];
+				google.protobuf.Int32Value purge_event_feed_after_days = 3;
```
Comment (lancewf) [inline api/config/event_feed/config_request.proto]: Adding the number of days to purge the event feed data to the Automate config.

File: components/event-feed-service/pkg/grpc/grpc.go [hunk 4]
```diff
@@ -66,6 +84,11 @@ func NewGRPCServer(connFactory *secureconn.Factory, c *config.EventFeed,
 
 	health.RegisterHealthServer(grpcServer, eventFeedServer.Health())
 
+	dataLifecycleServer := server.NewDataLifecycleManageableServer(
+		es_sidecar.NewEsSidecarClient(esSidecarConn), c.PurgeEventFeedAfterDays)
+
+	data_lifecycle.RegisterDataLifecycleManageableServer(grpcServer, dataLifecycleServer)
```
Comment (lancewf) [inline components/event-feed-service/pkg/grpc/grpc.go]: Starting the data lifecycle server.

File: components/event-feed-service/pkg/integration_test/feed_purge_test.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
+package integration_test
```
Comment (lancewf) [inline components/event-feed-service/pkg/integration_test/feed_purge_test.go]: Moving existing data lifecycle server test from the compliance event feed.

File: components/event-feed-service/pkg/integration_test/global_test.go [hunk 6]
```diff
@@ -38,6 +45,17 @@ func TestMain(m *testing.M) {
 	// run, e.g., initializing ES indices, inserting test data, etc.
 	testSuite.GlobalSetup()
 
+	// get the datalifecycle client so we can test the feed purge
```
Comment (lancewf) [inline components/event-feed-service/pkg/integration_test/global_test.go]: Creating a connection to the data lifecycle service.

File: .studio/golang [hunk 7]
```diff
@@ -184,6 +184,11 @@ function go_update_component() {
 # Adding auto tab complete
 complete -F _component_auto_complete go_update_component
 
+document "kill_running_service" <<DOC
+  kill a running component. This can be used to restart a component
+
+  example kill_running_service ingest-service
```
Comment (phiggins) [inline .studio/golang]: It'd be more clear what the command to run was if this was split onto multiple lines.
```suggestion
  Example:
      kill_running_service ingest-service
```

Comment (phiggins) [reply .studio/golang]: I'm not sure if we have a convention for this.

Comment (lancewf) [reply .studio/golang]: Thank you.

Comment (lancewf) [reply .studio/golang]: I am not sure why but that broke the DCO check. :man_shrugging:

Comment (phiggins) [reply .studio/golang]: That's an unfortunate side effect of the suggestion feature and DCO check not working well together https://chefio.slack.com/archives/C0808K15Y/p1561751386240400

File: .expeditor/verify_private.pipeline.yml [hunk 8]
```diff
@@ -285,7 +285,9 @@ steps:
   - label: "[integration] event-feed-service"
     command:
       - . scripts/verify_setup.sh
-      - hab studio run "source scripts/verify_studio_init.sh && start_deployment_service && chef-automate dev deploy-some chef/event-feed-service --with-deps && event_feed_integration"
+      - hab studio run "source scripts/verify_studio_init.sh &&
+          start_deployment_service &&
+          chef-automate dev deploy-some chef/data-lifecycle-service --with-deps && event_feed_integration"
```
Comment (phiggins) [inline .expeditor/verify_private.pipeline.yml]: Super nitpicky but it seems weird to have only the last line have two commands joined with `&&`.

Comment (lancewf) [reply .expeditor/verify_private.pipeline.yml]: I did not notice that.

**Pull request merged**